File: code/sorting/bogo_sort/bogo_sort.go
//Part of Cosmos Project by OpenGenus Foundation
//Bogo Sort implementation on golang
//Written by Guilherme Lucas(guilhermeslucas)
package main

import (
    "fmt"
    "sort"
    "math/rand"
)

func shuffle(arr []int) []int {
	for i := range arr {
		j := rand.Intn(i + 1)
		arr[i], arr[j] = arr[j], arr[i]
	}

	return arr
}

func bogoSort (array []int) []int {
    for {
        if sort.IntsAreSorted(array) {
            return array
        }
        array = shuffle(array[:])
    }
}

func main() {
    var array = []int{1,5,8,2,6,9}
    sorted := bogoSort(array)
    fmt.Println("Sorted sequence is:", sorted)
}

File: code/sorting/src/bogo_sort/bogo_sort.c
/* Part of cosmos by OpenGenus Foundation  */

/*
* bogo_sort.c
* created by Riya
*/

#include <stdio.h>    
#include <stdlib.h>    

int
main()    
{        
    int num[10]={1, 4, 7, 5, 9, 2, 6, 3, 8, 0};        
    int i;            
    bogosort(num, 10);        
    printf("The array after sorting is:");        
    for (i = 0;i < 10;i++) {            
        printf("%d\n", num[i]);        
    } printf("\n");    
}         
int
is_sorted(int *a, int n)    
{        
    while ( --n >= 1 ) {        
        if ( a[n] < a[n-1] ) 
            return (0);    
    } 
    return (1);    
}    
void
shuffle(int *a, int n)    
{        
    int i, t, temp;        
    for (i = 0;i < n;i++) {            
        t = a[i];            
        temp = rand() % n;             
        a[i] = a[temp];            
        a[temp] = t;        
    }    
}    
void
bogosort(int *a, int n)    
{        
    while ( !is_sorted(a, n) )             
    shuffle(a, n);    
}
